### PR TITLE
PR for task one in arm sprint

### PR DESCRIPTION
**Pull request by Khedr05**
video link : https://github.com/sherifkhadr/SPRINTS_Automotive_Software_Bootcamp/tree/deliver_branch/07_Sprint_Twelve/00_Run_WD_Tester/01_Video

### CODE REVIEW

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+				 }
+			else if(PORT_PIN_LEVEL_LOW==ConfigPtr[u8_Pin_Counter].PortPinLevelValue)
+	       {
+         CLR_BIT(GPIO_DATA_REG(port_num),pin_num);
+		     }
+				 else
+				 {
+					 /*invalid*/
+				 }
+		switch(ConfigPtr[u8_Pin_Counter].PortPinOutputCurrent)
```
Comment (Hala-Abdelkader) [inline 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c]: Switch case should have only one break

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+// Define the 2D array for interrupt handlers
+
+extern InterruptHandler interruptHandlers[6][8];
+
+
+
+
+void Port_Init( Port_ConfigType* ConfigPtr);
```
Comment (Hala-Abdelkader) [inline 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h]: Each function should return error state

**Review by Hala-Abdelkader — CHANGES_REQUESTED** [review]

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c (hunk 1 above)
Comment (Khedr05) [reply 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c]: done now you can check again the new commit

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h (hunk 2 above)
Comment (Khedr05) [reply 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h]: done now you can check again the new commit

**Comment by Khedr05** [comment]
done now you can check again the new commit

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h (hunk 2 above)
Comment (Hala-Abdelkader) [reply 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.h]: Great, good job.

File: 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c (hunk 1 above)
Comment (Hala-Abdelkader) [reply 07_Sprint_Twelve/01_RGB_LED_CONTROL_V1.0/00_Project/RGB_LED_SEQUANCE_V1.0/GPIO/port.c]: Great, good job.

**Pull request merged**